packages/alert-banner/src/AlertBanner.ts

```diff
@@ -0,0 +1,146 @@
+    SpectrumElement,
+    TemplateResult,
+} from '@spectrum-web-components/base';
+import { property } from '@spectrum-web-components/base/src/decorators.js';
+import '@spectrum-web-components/button/sp-close-button.js';
+import '@spectrum-web-components/icons-workflow/icons/sp-icon-alert.js';
+import '@spectrum-web-components/icons-workflow/icons/sp-icon-info.js';
+import styles from './alert-banner.css.js';
+
+export type AlertBannerVariants = 'neutral' | 'info' | 'negative' | '';
+const VALID_VARIANTS = ['neutral', 'info', 'negative'];
+
+/**
+ * @element sp-alert-banner
+ */
+export class AlertBanner extends SpectrumElement {
+    public static override get styles(): CSSResultArray {
+        return [styles];
+    }
+
+    @property({ type: Boolean, reflect: true })
+    public open = false;
+
+    @property({ type: Boolean, reflect: true })
+    public dismissible = false;
+
+    /**
+     * The variant applies specific styling when set to `negative` or `info`;
+     * `variant` attribute is removed when it's passed an invalid variant.
+     *
+     * @param {String} variant
+     */
+    @property({ type: String })
+    public set variant(variant: AlertBannerVariants) {
+        if (variant === this.variant) {
+            return;
+        }
+        const oldValue = this.variant;
+        if (this._isValidVariant(variant)) {
+            this.setAttribute('variant', variant);
+            this._variant = variant;
+        } else {
+            this.removeAttribute('variant');
+            this._variant = '';
+        }
+        this.requestUpdate('variant', oldValue);
+    }
+
+    public get variant(): AlertBannerVariants {
+        return this._variant;
+    }
+
+    private _variant: AlertBannerVariants = '';
+
+    protected _isValidVariant(variant: string): boolean {
+        return VALID_VARIANTS.includes(variant);
+    }
+
+    protected renderIcon(variant: string): TemplateResult {
+        switch (variant) {
+            case 'info':
+                return html`
+                    <sp-icon-info
+                        label="Information"
+                        class="type"
+                    ></sp-icon-info>
+                `;
+            case 'negative':
+                return html`
+                    <sp-icon-alert label="Error" class="type"></sp-icon-alert>
+                `;
+            default:
+                return html``;
+        }
+    }
+
+    private _shouldClose(): void {
+        const applyDefault = this.dispatchEvent(
+            new CustomEvent('close', {
+                composed: true,
+                bubbles: true,
+                cancelable: true,
+            })
+        );
+        if (applyDefault) {
+            this.close();
+        }
+    }
+
+    public close(): void {
+        this.open = false;
+    }
+
+    private _handleKeydown(event: KeyboardEvent): void {
+        if (event.code === 'Escape' && this.dismissible) {
+            this._shouldClose();
+        }
+    }
```

Comment:
Should this really be a thing, or should be document that consumers should be opening this as an Overlay?

If we should be doing this, it's likely better to keep the event binding only when `open` so that the default `<sp-alert-banner>` element does not do unneeded work.

Reply:
I added this functionality because it was defined in Spectrum's docs and also included in the [RFC](https://github.com/adobe/spectrum-web-components/discussions/4106). Following your suggestion, the keydown listener is added only when open is true.